expr/expr.go

```diff
@@ -122,20 +141,40 @@ func (e expr) consumeBasicArg(pos int, exp Arg) (int, error) {
 		}
 		return pos, nil
 	case ArgFloat:
+		switch got.etype {
 		// integer is also a valid float, just happened to have no decimals
-		if got.etype != etFloat && got.etype != etInt {
-			return 0, ErrBadArgumentStr{"float", got.etype.String()}
-		}
-		for _, va := range v.validator {
-			if err := va(got); err != nil {
-				return 0, generateValidatorError(v.key, err)
+		case etInt, etFloat:
+			for _, va := range v.validator {
+				if err := va(got); err != nil {
+					return 0, generateValidatorError(v.key, err)
+				}
 			}
+			if got.etype == etInt {
+				*v.val = float64(got.int)
+			} else {
+				*v.val = got.float
+			}
+		case etString:
+			// allow "2.5", '5.4', etc. some people put quotes around their floats.
+			f, ok := strToFloat(got.str)
+			if !ok {
+				return 0, ErrBadArgumentStr{"int", got.etype.String()}
+			}
+			gotFake := expr{
+				str:   got.str,
+				float: f,
+				etype: etInt,
```

Comment:
would this ever need to be `etFloat`?

Reply:
oops, it should be etFloat yeah.
we only construct these for the validators which take expr values but typically only use the 'float' field inside of it.
but may as well be correct!